File: resources/expected/ros2/isolette/lax/src/isolette_cpp_pkg/src/base_code/heat_source_cpi_heat_controller_base_src.cpp
#include "isolette_cpp_pkg/base_headers/heat_source_cpi_heat_controller_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

heat_source_cpi_heat_controller_base::heat_source_cpi_heat_controller_base() : Node("heat_source_cpi_heat_controller")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    heat_source_cpi_heat_controller_heat_control_subscription_ = this->create_subscription<isolette_cpp_pkg_interfaces::msg::OnOff>(
        "heat_source_cpi_heat_controller_heat_control",
        1,
        std::bind(&heat_source_cpi_heat_controller_base::handle_heat_control, this, std::placeholders::_1), subscription_options_);

    heat_source_cpi_heat_controller_heat_out_publisher_ = this->create_publisher<isolette_cpp_pkg_interfaces::msg::Heat>(
        "heat_source_cpi_heat_controller_heat_out",
        1);

    // timeTriggered callback timer
    periodTimer_ = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&heat_source_cpi_heat_controller_base::timeTriggered, this), cb_group_);

}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void heat_source_cpi_heat_controller_base::handle_heat_control(const isolette_cpp_pkg_interfaces::msg::OnOff::SharedPtr msg)
{
    heat_control_msg_holder = msg;
}

isolette_cpp_pkg_interfaces::msg::OnOff::SharedPtr heat_source_cpi_heat_controller_base::get_heat_control() {
    return heat_control_msg_holder;
}

void heat_source_cpi_heat_controller_base::put_heat_out(isolette_cpp_pkg_interfaces::msg::Heat msg)
{
    heat_source_cpi_heat_controller_heat_out_publisher_->publish(msg);
}

File: resources/expected/ros2/python-isolette_inverted_topics/lax/src/isolette_py_pkg/isolette_py_pkg/base_code/heat_source_cpi_heat_controller_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from collections import deque
from isolette_py_pkg.user_code.heat_source_cpi_heat_controller_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from isolette_py_pkg_interfaces.msg import OnOff
from isolette_py_pkg_interfaces.msg import Heat

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class heat_source_cpi_heat_controller_base(Node):
    def __init__(self):
        super().__init__("heat_source_cpi_heat_controller")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.heat_source_cpi_heat_controller_heat_control_subscription_ = self.create_subscription(
            OnOff,
            "thermostat_regulate_temperature_manage_heat_source_mhst_heat_control",
            self.handle_heat_control,
            1,
            callback_group=self.cb_group_)

        self.heat_source_cpi_heat_controller_heat_out_publisher_ = self.create_publisher(
            Heat,
            "heat_source_cpi_heat_controller_heat_out",
            1)

        # timeTriggered callback timer
        self.periodTimer_ = self.create_timer(1, self.timeTriggered, callback_group=self.cb_group_)

        self.heat_control_msg_holder = None

    def init_heat_control(self, val):
        self.heat_control_msg_holder = val

    def timeTriggered(self):
        raise NotImplementedError("Subclasses must implement this method")

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def handle_heat_control(self, msg):
        self.heat_control_msg_holder = msg

    def get_heat_control(self):
        return self.heat_control_msg_holder

    def put_heat_out(self, msg):
        self.heat_source_cpi_heat_controller_heat_out_publisher_.publish(msg)

